src/gflownet/envs/mol_building_env.py

```diff
@@ -201,7 +238,22 @@ def GraphAction_to_aidx(self, g: gd.Data, action: GraphAction) -> Tuple[int, int
             col = (
                 self.bond_attr_values[action.attr].index(action.value) - 1 + self.bond_attr_logit_slice[action.attr][0]
             )
-        type_idx = self.action_type_order.index(action.action)
+        elif action.action is GraphActionType.RemoveNode:
+            row = action.source
+            col = 0
+        elif action.action is GraphActionType.RemoveNodeAttr:
+            row = action.source
+            col = self.settable_atom_attrs.index(action.attr)
+        elif action.action is GraphActionType.RemoveEdge:
+            row = ((g.edge_index.T == torch.tensor([(action.source, action.target)])).prod(1)).argmax()
+            row = int(row) // 2  # edges are duplicated, but edge logits are not
```

Comment:
I  don't really get this line `row = int(row) // 2`

doesn't line 248 give you the index of the directed edge to remove ? Why take `row // 2` again here ?
Also on the above, you are adding two edges per bond right ? and removing both simultaneously or one at a time ?

Reply:
torch geometric assumes all edges are directed, so to have an undirected graph we must encode the edge list as `edge_index = [[a,b], [b,a], [c,d], [d,c], ...]`. If we have `n` (undirected) edges, we only ask the model to produce `n` logits, which is why in `aidx_to_GraphAction` we do `act_row * 2` to retrieve the edge endpoints in `g.edge_index`.
So yes in some sense we remove both simultaneously, from the point of view of torch geometric.

Reply:
Yes, I understand that part. I was confused by the `*2` since there would need to be a specific ordering in the `edge_index` matrix to fetch the right couple of nodes. I am assuming from your example, that for any bond, the two corresponding directed edges `(u,v), (v,u)` will always be contiguous in `edge_index`.

Reply:
Ah good point, yes it is the case that they are contiguous. I'll update the comments.